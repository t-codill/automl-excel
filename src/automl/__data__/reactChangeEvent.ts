// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
