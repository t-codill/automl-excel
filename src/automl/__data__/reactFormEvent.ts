// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
