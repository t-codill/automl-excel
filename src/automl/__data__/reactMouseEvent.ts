// tslint:disable-next-line: no-any
export const reactMouseEvent = {} as unknown as React.MouseEvent<any>;
