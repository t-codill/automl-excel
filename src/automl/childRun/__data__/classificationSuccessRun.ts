import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const classificationSuccessRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_001",
        parentRunId: "AutoML_000",
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: {
            iteration: "9",
            run_algorithm: "Ensemble",
            run_preprocessor: "rpp1"
        }
    },
    runMetrics: {
        scalarMet1: "1",
        scalarMet2: "2",
        scalarMet3: "3",
        accuracy_table: {},
        confusion_matrix: {}
    },
    modelUri: "sampleModelUri",
    experimentName: "classificationSuccess"
};
