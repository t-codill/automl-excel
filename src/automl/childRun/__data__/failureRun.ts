import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const failureRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_002",
        status: "Failed",
        parentRunId: "AutoML_000",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: undefined
    },
    runMetrics: {},
    experimentName: ""
};
