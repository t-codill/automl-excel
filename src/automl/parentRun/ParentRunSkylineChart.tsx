//import { SkylineChart } from "@mlworkspace/charts";
import { map } from "lodash";
import { MessageBar } from "office-ui-fabric-react";
import * as React from "react";
import { generateRunName } from "../common/utils/generateRunName";
// import { isTerminatedRunStatus } from "../common/utils/run";
import { safeParseJson } from "../common/utils/safeParseJson";
import { BaseComponent } from "../components/Base/BaseComponent";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { IParentRunData } from "./ParentRun";

export class ParentRunSkylineChart extends BaseComponent<IParentRunData, {}, {}> {
    protected serviceConstructors = {};
    protected getData = undefined;

    public readonly render = (): React.ReactNode => {
        if (!this.props.childRunMetrics || !this.props.run || !this.props.run.properties) {
            return <PageLoadingSpinner />;
        }

        if (this.props.childRunMetrics.length < 1
        ) {
            return (
                <MessageBar>No data for visualization</MessageBar>
            );
        }

        // if (isTerminatedRunStatus(this.props.run.status)) {
        //     // TODO: render shimmer
        // )

        const min = safeParseJson(this.props.run.properties.AMLSettingsJsonString).metric_operation === "minimize";
        const names = map(this.props.childRuns, generateRunName);
        //@ts-ignore
        const skylineChartInput = {
            metrics: this.props.childRunMetrics,
            names,
            primaryMetricName: this.props.run.properties.primary_metric,
            min
        };
        return null /*(<SkylineChart
            metric={skylineChartInput}
            theme={this.context.theme}
            configOverride={{
                layout: {
                    margin: {
                        l: 60,
                        r: 60,
                        b: 60,
                        t: 40
                    },
                },
                useResizeHandler: true,
                config: {
                    displayModeBar: false
                }
            }} />)*/;
    }
}
