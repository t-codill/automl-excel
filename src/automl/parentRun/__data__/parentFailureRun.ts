import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IParentRunData } from "../ParentRun";

export const parentFailureRun: NotNullableProperties<IParentRunData> = {
    run: {
        runId: "AutoML_000",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: "",
            primary_metric: "pm",
            errors: "Setup iteration failed: \\n \t\t\t\t\texception: \\n \t\t\t\t\t\tUnforseenConsequence('Invalid subject in the test chamber') \\n"
        }
    },
    experimentName: "Test",
    childRuns: [],
    childRunMetrics: [
        {
            spearman_correlation: 0.877770827248834
        }
    ]
};
