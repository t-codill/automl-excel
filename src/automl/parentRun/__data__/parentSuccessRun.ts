import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IParentRunData } from "../ParentRun";

export const parentSuccessRunAMLSettings = {
  primary_metric: "AUC_weighted",
  data_script: null,
  compute_target: "local",
  n_cross_validations: 3,
  max_cores_per_iteration: 1,
  blacklist_algos: [],
  exit_score: 100,
  name: "my_experiment",
  task_type: "classification"
};
export const parentSuccessRun: NotNullableProperties<IParentRunData> = {
  run: {
    runId: "AutoML_000",
    parentRunId: undefined,
    status: "Completed",
    startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
    endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
    properties: {
      AMLSettingsJsonString: JSON.stringify(parentSuccessRunAMLSettings)
    },
    tags: {
      iterations: "2"
    },
  },
  experimentName: "Test",
  childRuns: [
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_001",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      experimentId: "exp",
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_001",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_algorithm: "RandomForest",
        run_preprocessor: "MaxAbsScaler",
        score: "0.977770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_002",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_002",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_preprocessor: "MaxAbsScaler",
        score: "0.877770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_003",
      parentRunId: "AutoML_000",
      status: "Canceled",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_003",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        score: "0.877770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_004",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_004",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_algorithm: "MaxAbsScaler",
        score: "0.827770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      },
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_005",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_005",
    }
  ],
  childRunMetrics: [
    {
      spearman_correlation: 0.877770827248834
    }
  ]
};
