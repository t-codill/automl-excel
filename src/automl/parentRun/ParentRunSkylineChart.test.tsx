import { shallow } from "enzyme";
import * as React from "react";
import { parentSuccessRun } from "./__data__/parentSuccessRun";
import { ParentRunSkylineChart } from "./ParentRunSkylineChart";

describe("Parent Run Skyline Chart", () => {

    it("should render loading image when no run", () => {
        const tree = shallow(<ParentRunSkylineChart
            experimentName="foo"
            run={undefined}
            childRuns={undefined}
            childRunMetrics={undefined}
        />);
        expect(tree)
            .toMatchInlineSnapshot("<pageLoadingSpinner />");
    });
    it("should render No Data when no run", () => {
        const tree = shallow(<ParentRunSkylineChart
            experimentName="foo"
            run={parentSuccessRun.run}
            childRuns={[]}

            childRunMetrics={[]}
        />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render chart when successful run", () => {
        const tree = shallow(<ParentRunSkylineChart
            experimentName="foo"
            run={parentSuccessRun.run}
            childRuns={parentSuccessRun.childRuns}
            childRunMetrics={parentSuccessRun.childRunMetrics}
        />);
        expect(tree)
            .toMatchSnapshot();
    });
});
