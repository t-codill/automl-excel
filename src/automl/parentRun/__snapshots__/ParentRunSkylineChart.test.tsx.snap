// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent Run Skyline Chart should render No Data when no run 1`] = `
<StyledMessageBarBase>
  No data for visualization
</StyledMessageBarBase>
`;

exports[`Parent Run Skyline Chart should render chart when successful run 1`] = `
<e
  configOverride={
    Object {
      "config": Object {
        "displayModeBar": false,
      },
      "layout": Object {
        "margin": Object {
          "b": 60,
          "l": 60,
          "r": 60,
          "t": 40,
        },
      },
      "useResizeHandler": true,
    }
  }
  metric={
    Object {
      "metrics": Array [
        Object {
          "spearman_correlation": 0.877770827248834,
        },
      ],
      "min": false,
      "names": Array [
        "MaxAbsScaler, RandomForest",
        "MaxAbsScaler",
        "<N/A>",
        "MaxAbsScaler",
        "<N/A>",
      ],
      "primaryMetricName": undefined,
    }
  }
  theme="light"
/>
`;
